Tests/test_SeqFeature.py

```diff
@@ -337,6 +337,26 @@ def test_reference_in_compound_location_sequence(self):
         self.assertEqual(type(sequence), Seq.Seq)
         self.assertEqual(sequence, "ccaatgg")
 
+    def test_origin_spanning_location(self):
+        """Test location spanning origin."""
+        # Regular origin-spanning sequence
+        self.assertEqual(
+            str(SimpleLocation.fromstring("4..2", 4, True)), "join{[3:4], [0:2]}"
+        )
+        self.assertEqual(
+            str(SimpleLocation.fromstring("complement(4..2)", 4, True)),
+            "join{[0:2](-), [3:4](-)}",
+        )
+
+        # Origin-spanning location containing the entire sequence
+        self.assertEqual(
+            str(SimpleLocation.fromstring("3..2", 4, True)), "join{[2:4], [0:2]}"
+        )
+        self.assertEqual(
+            str(SimpleLocation.fromstring("complement(3..2)", 4, True)),
+            "join{[0:2](-), [2:4](-)}",
+        )
+
```

Comment:
Yay -  that looks like a simple but comprensive test based on your motivating example.